### PR TITLE
Added unit test to main engine

### PR DESCRIPTION
**Pull request by paulvazbad**
Refractor the main_engine package to include the code in a python module inside called main_engine.
Modified CMakeLists to include a test script.
Added rosunit as a dependency on packages.xml

Modified both files to include integration tests in the future.

In a follow up PR: Include useful unit tests, so far I only have a placeholder test to test the initial state of the list in the ActionQueue. However, I don't want this PR to become too big.

To run the test:
`catkin_make run_tests`

### CODE REVIEW

File: catkin_home/src/main_engine/src/main_engine/Action.py [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+        self.specific_function.setdefault("default")
+
+    def specific_function_not_found(self):
+        print("Not found the corresponding action")
+
+    def print_self(self, endline=False):
+        if(endline == False):
+            print(self.get_description(),)
+            return
+        print(self.get_description())
```
Comment (tomvik) [inline catkin_home/src/main_engine/src/main_engine/Action.py:39]: what's the point of the if?

Comment (paulvazbad) [reply catkin_home/src/main_engine/src/main_engine/Action.py]: If i want to print them all in the same line I call the function without endline.

**Comment by tomvik** [comment]
LGTM! But first, can you please make an experiment? Make a push with a test that you know will fail, and let's verify that the workflow fails.

**Comment by tomvik** [comment]
@paulvazbad Is this good to merge? On a second PR I'll update the catkin.yaml file and add a script to break the build on failing tests

**Pull request merged**